### PR TITLE
Enable pwm for beagleconnect freedom

### PR DESCRIPTION
**Pull request by Ayush1325**
- Add channel, flags to pwm-cells
- Replace __ASSERT_UNREACHABLE with CODE_UNREACHABLE
- Add and test pwm for beagleconnect freedom

### CODE REVIEW

File: dts/bindings/pwm/ti,cc13xx-cc26xx-timer-pwm.yaml [hunk 1]
```diff
@@ -69,4 +69,6 @@ properties:
     required: true
 
 pwm-cells:
-- period
+  - channel
+  - period
```
Comment (decsny) [inline dts/bindings/pwm/ti,cc13xx-cc26xx-timer-pwm.yaml:73]: why is this needed, since period can already be controlled by the PWM API?

Comment (Ayush1325) [reply dts/bindings/pwm/ti,cc13xx-cc26xx-timer-pwm.yaml:73]: [`PWM_DT_SPEC_GET`](https://docs.zephyrproject.org/latest/hardware/peripherals/pwm.html#c.PWM_DT_SPEC_GET) and related macros (used in the example as well as a lot of external zephyr code like Arduino module for zephyr) consider channel and period to be required properties.

Also, period was already present, I added channel.

Comment (henrikbrixandersen) [reply dts/bindings/pwm/ti,cc13xx-cc26xx-timer-pwm.yaml:73]: This looks correct to me.

File: samples/basic/blinky_pwm/boards/beagleconnect_freedom.overlay
Comment (henrikbrixandersen) [inline samples/basic/blinky_pwm/boards/beagleconnect_freedom.overlay:1]: Why not add these directly to the board DTS?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: samples/basic/blinky_pwm/boards/beagleconnect_freedom.overlay
Comment (Ayush1325) [reply samples/basic/blinky_pwm/boards/beagleconnect_freedom.overlay:1]: These pins are part of the mikrobus connector. So using them separately like this is not the primary purpose.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by vaishnavachath — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Review by kartben — APPROVED** [review]
